source/index.js

```diff
@@ -54,9 +54,7 @@
 			readdir: function(path, callback) {
 				client.getDir(endpoint, path)
 					.then(function(dirData) {
-						(callback)(null, dirData.map(function(dirEntry) {
-							return dirEntry.basename;
-						}));
+						(callback)(null, dirData);
```

Comment:
Ok, a new method would fit better, but it is necesary, because webdav protocol hasn't the performance to stat tons of files in low time, so sometimes (like owncloud) you can't stat all files every time you want to list a directory.